test/functional/page_objects/visualize_page.js

```diff
@@ -391,27 +391,8 @@ export function VisualizePageProvider({ getService, getPageObjects }) {
       return await element.getVisibleText();
     }
 
-    // clickBucket(bucketType) 'X-Axis', 'Split Area', 'Split Chart'
-    async clickBucket(bucketName, type = 'bucket') {
-      const testSubject = type === 'bucket' ? 'bucketsAggGroup' : 'metricsAggGroup';
-      await retry.try(async () => {
-        const chartTypes = await retry.try(
-          async () => await find.allByCssSelector(`[data-test-subj="${testSubject}"] .list-group-menu-item`));
-        log.debug('found bucket types ' + chartTypes.length);
-
-        async function getChartType(chart) {
-          const chartString = await chart.getVisibleText();
-          if (chartString === bucketName) {
-            await chart.click();
-            return true;
-          }
-        }
-        const getChartTypesPromises = chartTypes.map(getChartType);
-        const clickResult = await Promise.all(getChartTypesPromises);
-        if (!clickResult.some(result => result === true)) {
-          throw new Error(`bucket ${bucketName} not found`);
-        }
-      });
+    async clickBucket(bucketName, type = 'buckets') {
+      await find.clickByCssSelector(`[group-name="${type}"] [data-test-subj="${bucketName}"]`);
```

Comment:
also, is there a logical explanation how this change resolves those flaky tests? or what does the new `clickBucket` method do differently from the old one that will improve the situation ?